packages/common/solana/utils.ts

```diff
@@ -278,49 +348,56 @@ export async function enrichAtaParams(connection: Connection, paramsBatch: AtaPa
  * @param connection - Solana client connection
  * @param payer - Transaction invoker, should be a signer
  * @param paramsBatch - Array of Params for an each ATA account: {mint, owner}
+ * @param commitment - optional commitment that will be used to fetch Blockhash
  * @returns Unsigned Transaction with create ATA instructions
  */
 export async function generateCreateAtaBatchTx(
   connection: Connection,
   payer: PublicKey,
   paramsBatch: AtaParams[],
+  commitment?: Commitment,
 ): Promise<{
-  tx: Transaction;
+  tx: VersionedTransaction;
   hash: BlockhashWithExpiryBlockHeight;
+  context: Context;
 }> {
   paramsBatch = await enrichAtaParams(connection, paramsBatch);
   const ixs: TransactionInstruction[] = await Promise.all(
     paramsBatch.map(async ({ mint, owner, programId }) => {
       return createAssociatedTokenAccountInstruction(payer, await ata(mint, owner), owner, mint, programId);
     }),
   );
-  const hash = await connection.getLatestBlockhash();
-  const tx = new Transaction({
-    feePayer: payer,
-    blockhash: hash.blockhash,
-    lastValidBlockHeight: hash.lastValidBlockHeight,
-  }).add(...ixs);
-  return { tx, hash };
+  const { value: hash, context } = await connection.getLatestBlockhashAndContext({ commitment });
+  const messageV0 = new TransactionMessage({
+    payerKey: payer,
+    recentBlockhash: hash.blockhash,
+    instructions: ixs,
+  }).compileToV0Message();
+  const tx = new VersionedTransaction(messageV0);
+  return { tx, hash, context };
 }
 
 /**
  * Creates ATA for an array of owners
  * @param connection - Solana client connection
  * @param invoker - Transaction invoker and payer
  * @param paramsBatch - Array of Params for an each ATA account: {mint, owner}
+ * @param commitment - optional commitment that will be used to fetch Blockhash
  * @returns Transaction signature
  */
 export async function createAtaBatch(
   connection: Connection,
   invoker: Keypair | SignerWalletAdapter,
   paramsBatch: AtaParams[],
+  commitment?: Commitment,
 ): Promise<string> {
-  const { tx, hash } = await generateCreateAtaBatchTx(
+  const { tx, hash, context } = await generateCreateAtaBatchTx(
     connection,
     invoker.publicKey!,
     await enrichAtaParams(connection, paramsBatch),
+    commitment,
   );
-  return signAndExecuteTransaction(connection, invoker, tx, hash);
+  return signAndExecuteTransaction(connection, invoker, tx, { hash, context, commitment });
```

Comment:
We should definitely split utils into meaningful files :) I guess for one of next PRs